### PR TITLE
Dashboard: Allows the d-solo route to be used without slug

### LINKED ISSUE #11617: Embedded URL in 5.0 requires dashboard name
Grafana 5.0.3
MetricTank

We are trying to use an embedded URL:
https://grafana.prod.bloomberg.com/d-solo/LPlyrOZmz/machine-view?orgId=1&var-Machine=njlxdev1&from=1523873681815&to=1523916881815&panelId=2

This works. ... however if we strip out the dashboard name it doesn't work.

https://grafana.prod.bloomberg.com/d-solo/LPlyrOZmz?orgId=1&var-Machine=njlxdev1&from=1523873681815&to=1523916881815&panelId=2 

The GUID should be sufficient.

### PR DESCRIPTION
**Pull request by 97amarnathk**
Removes the `:slug` fromt the routes and the API. This allows the Embedded URLs to work without requiring the dashboard name.

**Which issue(s) this PR fixes**:

Fixes #11617

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Get("/dashboard/db/:slug", reqSignedIn, redirectFromLegacyDashboardURL, hs.Index)
 	r.Get("/dashboard/script/*", reqSignedIn, hs.Index)
 	r.Get("/dashboard-solo/snapshot/*", hs.Index)
-	r.Get("/d-solo/:uid/:slug", reqSignedIn, hs.Index)
```
Comment (torkelo) [inline pkg/api/api.go]: this is a breaking change (I think, have you tried urls with slug?) try making slug optional

Comment (marefr) [reply pkg/api/api.go]: @97amarnathk I tried to explain in https://github.com/grafana/grafana/issues/11617#issuecomment-537434630 that you need to add new routes and keep the existing ones (that includes slug-param. So the changes you've made now is what we want, but we still want the existing routes to be kept intact. Does this makes sense to you? Thanks

Comment (97amarnathk) [reply pkg/api/api.go]: @marefr, So I need to keep the slug URLs too (instead of deleting them)?

Comment (97amarnathk) [reply pkg/api/api.go]: I have revised the PR.

File: public/app/routes/routes.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export function setupAngularRoutes($routeProvider: route.IRouteProvider, $locati
         component: importDashboardPage,
       },
     })
-    .when('/d/:uid/:slug', {
+    .when('/d/:uid', {
```
Comment (marefr) [inline public/app/routes/routes.ts]: This is the wrong route to update. Need to copy line 73 to 84 and paste as new line after line 73 where the route should be updated to `/d-solo/:uid`. Makes sense?

**Review by marefr — CHANGES_REQUESTED** [review]
Some changes still needed.

File: public/app/routes/routes.ts (hunk 2 above)
Comment (97amarnathk) [reply public/app/routes/routes.ts]: Oh I got it. I have updated the PR.

**Review by marefr — APPROVED** [review]
Great. Works as expected

**Comment by marefr** [comment]
@97amarnathk Thank you for contributing to Grafana! 

Totally optional, but may I ask you if you're participating in [Hacktoberfest](https://hacktoberfest.digitalocean.com/) and whether you've find this issue through the hacktoberfest label? Trying to find out if new contributors are finding Grafana due to Hacktoberfest.

**Comment by 97amarnathk** [comment]
@marefr 
Actually I started reading up on Go/Prometheus, and therefore Grafana came in my Github repositories reccomendation. Added to this, Hacktoberfest provided me motivation to get started with Grafana.

P.S. Can I get an issue to solve ? Something which is a bit more involved but has a sufficient enough learning curve for a beginner in Grafana and Go?

**Comment by marefr** [comment]
@97amarnathk thanks, that's great :smile: 

Maybe you can have a look [here ](https://github.com/grafana/grafana/issues?q=is%3Aopen+is%3Aissue+backend+label%3A%22help+wanted%22) or [here](https://github.com/grafana/grafana/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+) for issues that's labeled help wanted. Write a comment in an issue you would like to work on or are interested in and we'll get back to you with some additional pointers.

**Pull request merged**